plugins/chat/lib/message_mover.rb

```diff
@@ -16,6 +16,19 @@
 # all of the references associated to a chat message (e.g. reactions, bookmarks,
 # notifications, revisions, mentions, uploads) will be updated to the new
 # message IDs via a moved_chat_messages temporary table.
+#
+# Reply chains are a little complex. No reply chains are preserved when moving
+# messages into a new channel. Remaining messages that referenced moved ones
+# have their in_reply_to_id cleared so the data makes sense.
+#
+# Threads are even more complex. No threads are preserved when moving messages
+# into a new channel, they end up as just a flat series of messages that are
+# not in a chain. If the original message of a thread and N other messages
+# in that thread, then any messages left behind just get placed into a new
+# thread. Message moving will be disabled in the thread UI while
+# enable_experimental_chat_threaded_discussions is present, its too complicated
+# to have end users reason about for now, and we may want a standalone
+# "Move Thread" UI later on.
```

Comment:
I wonder if we don't tweak instructions of `plugins/chat/assets/javascripts/discourse/components/chat-message-move-to-channel-modal-inner.hbs` to contain a word about this.

Reply:
Okay will take a look

Reply:
I won't mention threads for now but will put a TODO there to add that later, will add a note about flattening replies.